packages/telemetry/src/index.ts

```diff
@@ -0,0 +1,241 @@
+
+      // Start span with parent context.
+      span = this.tracer.startSpan(name, opts, ctx)
+    }
+
+    // Save off span, duplicate names currently not handled.
+    this.spans[name] = span
+
+    // If this is an active span, set it as the new active span
+    if (active) {
+      const _end = span.end
+
+      // override the end function to allow us to pop the span off the queue if found.
+      span.end = (endTime) => {
+        // find the span in the queue by spanId
+        const index = this.activeSpanQueue.findIndex((element: Span) => {
+          return element.spanContext().spanId === span.spanContext().spanId
+        })
+
+        // if span exists, remove it from the queue
+        if (index > -1) {
+          this.activeSpanQueue.splice(index, 1)
+        }
+
+        _end.call(span, endTime)
+      }
+
+      this.activeSpanQueue.push(span)
+    }
+
+    return span
+  }
+
+  /**
+   * Return requested span
+   * @param name - span name to retrieve
+   * @returns Span | undefined
+   */
+  getSpan (name: string): Span | undefined {
+    return this.spans[name]
+  }
+
+  /**
+   * Search the span queue for the active span that meets the criteria
+   * @param fn - function to search the active spans
+   * @returns Span | undefined
+   */
+  findActiveSpan (fn: findActiveSpan): Span | undefined {
+    return (this.activeSpanQueue as Array<SpanPlus>).find(fn)
+  }
+
+  /**
+   * Ends specified active span and any active child spans
+   * @param span - span to end
+   */
+  endActiveSpanAndChildren (span?: Span | void) {
+    if (!span) {
+      return
+    }
+
+    const startIndex = this.activeSpanQueue.findIndex((element: Span) => {
+      return element.spanContext().spanId === span.spanContext().spanId
+    })
+
+    this.activeSpanQueue.slice(startIndex).forEach((spanToEnd) => {
+      span.setAttribute('spanEndedPrematurely', true)
+      spanToEnd?.end()
+    })
+  }
+
+  /**
+   * Returns the context object for the active span.
+   * @returns the context
+   */
+  getActiveContextObject (): contextObject {
+    const rootSpan = this.activeSpanQueue[this.activeSpanQueue.length - 1]
+
+    // If no root span, nothing to return
+    if (!rootSpan) {
+      return {}
+    }
+
+    const ctx = openTelemetry.trace.setSpan(openTelemetry.context.active(), rootSpan)
+    let myCtx = {}
+
+    openTelemetry.propagation.inject(ctx, myCtx)
+
+    return myCtx
+  }
+
+  /**
+   * Shuts down telemetry and flushes any batched spans.
+   * @returns promise
+   */
+  shutdown (): Promise<void> {
+    return this.provider.shutdown()
+  }
+
+  /**
+   * Returns the assigned exporter
+   * @returns SpanExporter
+   */
+  getExporter (): SpanExporter {
+    return this.exporter
+  }
+}
+
+/**
+ * The telemetry Noop class is used when telemetry is disabled.
+ * It should mirror all the existing functions in telemetry, but no-op for
+ * all operations.
+ */
+export class TelemetryNoop {
```

Comment:
Suggest defining the telemetry interface so both `TelemetryNoop` and `Telemetry` classes can implement the interface.